tests/change_dump_date_test.py

```diff
@@ -0,0 +1,359 @@
+         True),
+    ]
+)
+
+def test_parse_args(monkeypatch, user_args, expected_namespace, should_raise):
+    """
+    Test parse_args() function for different command-line arguments.
+    """
+    monkeypatch.setattr("sys.argv", ["script_name"] + user_args)
+
+    if should_raise:
+        with pytest.raises(ValueError):  # Ensuring validation rules trigger exceptions
+            parse_args()
+    else:
+        parsed_args = parse_args()
+        expected = argparse.Namespace(**expected_namespace)
+        assert vars(parsed_args) == vars(expected)
+
+# Test the file header date change
+class MockFieldsFile:
+    def __init__(self):
+        self.fixed_length_header = MockHeader()
+class MockHeader:
+    def __init__(self):
+        self.t1_year = 2000
+        self.v1_year = 2000
+        self.t1_month = 1
+        self.v1_month = 1
+        self.t1_day = 1
+        self.v1_day = 1
+
+# Parameterized test for the function
+@pytest.mark.parametrize(
+    "new_year, new_month, new_day, expected_year, expected_month, expected_day",
+    [
+        (2025, 12, 31, 2025, 12, 31),  # Test case 1: Set full date
+        (2025, None, None, 2025, 1, 1),  # Test case 2: Only year set
+        (None, 5, None, 2000, 5, 1),  # Test case 3: Only month set
+        (None, None, 15, 2000, 1, 15),  # Test case 4: Only day set
+        (None, None, None, 2000, 1, 1),  # Test case 5: No changes
+    ]
+)
+def test_change_header_date_file(new_year, new_month, new_day, expected_year, expected_month, expected_day):
+    """
+    This test check that the fileheader is being changed properly depending on which parameters it is given.
+    """
+
+    ff = MockFieldsFile()
+
+    change_header_date_file(ff, new_year, new_month, new_day)
+
+    assert ff.fixed_length_header.t1_year == expected_year
+    assert ff.fixed_length_header.v1_year == expected_year
+    assert ff.fixed_length_header.t1_month == expected_month
+    assert ff.fixed_length_header.v1_month == expected_month
+    assert ff.fixed_length_header.t1_day == expected_day
+    assert ff.fixed_length_header.v1_day == expected_day
+
+# Mocking the fields and the FieldsFile
+class MockField:
+    def __init__(self):
+        self.lbyr = 2000  # Default year
+        self.lbmon = 1    # Default month
+        self.lbdat = 1    # Default day
+class MockFieldsTestFile:
+    def __init__(self, num_fields=1):
+        self.fields = [MockField() for _ in range(num_fields)]  # Create a list of fields
+
+# Parameterized test for the function
+@pytest.mark.parametrize(
+    "new_year, new_month, new_day, expected_year, expected_month, expected_day",
+    [
+        (2025, 12, 31, 2025, 12, 31),  # Test case 1: Set full date
+        (2025, None, None, 2025, 1, 1),  # Test case 2: Only year set
+        (None, 5, None, 2000, 5, 1),  # Test case 3: Only month set
+        (None, None, 15, 2000, 1, 15),  # Test case 4: Only day set
+        (None, None, None, 2000, 1, 1),  # Test case 5: No changes
+    ]
```

Comment:
```suggestion
    [
        (2025, 12, 31, 2025, 12, 31),  # Test case 1: Set full date
        (2025, None, None, 2025, 2, 12),  # Test case 2: Only year set
        (None, 5, None, 1993, 5, 12),  # Test case 3: Only month set
        (None, None, 15, 1993, 2, 15),  # Test case 4: Only day set
        (None, None, None, 1993, 2, 12),  # Test case 5: No changes
    ]
```
Made it consistent with changes below